[ZT] WARP QR code enrollment (#20904)
cloudflare/cloudflare-docs: src/content/docs/cloudflare-one/connections/connect-devices/warp/deployment/manual-deployment.mdx

## iOS, Android, and ChromeOS

### Enroll manually

<Render file="warp/enroll-ios-android" />

The device is now protected by your organization's Zero Trust policies.

### Enroll using a QR code

<Details header="Feature availability">

| System   | Availability | Minimum WARP version |
| -------- | ------------ | -------------------- |
| Windows  | ❌           |                       |
| macOS    | ❌           |                      |
| Linux    | ❌           |                      |

> | iOS      | ✅           | 1.5                   |

Author: Confirm supported WARP versions

Reviewer: Mobile clients do not support this feature for 1.1.1.1 app. We only support for Cloudflare One Agent app. 

Cloudflare One Agent app version which supports this feature are listed below:

iOS - 1.10 - https://jira.cfdata.org/browse/CLIENT-11028
Android - 2.4 - https://jira.cfdata.org/browse/CLIENT-11029

| Android  | ✅           |                      |
| ChromeOS | ✅           |                      |

</Details>

Administrators can provide users with a custom login link or QR code that automatically fills in your organization's <GlossaryTooltip term="team name">team name</GlossaryTooltip> during device enrollment. Using a QR code reduces the potential for error that comes with manual entry of the team name.

#### Generate a QR code

To generate a device enrollment QR code:

1. Copy the following link, replacing `your_team_name` with your Zero Trust <GlossaryTooltip term="team name">team name</GlossaryTooltip>:
		```txt
		cf1app://oneapp.cloudflare.com/team?name=your_team_name

> 		```

Reviewer: `cf1app://oneapp.cloudflare.com/team?name=your_team_name` is not required for users. Infact, users don't care about this.

But yes, admin should follow the URL and replace `your_team_name` with their ZT org name and generate the QR code using this URL.

2. Use any QR code generator to embed the link in a QR code.
3. Distribute the QR code to users.

#### Use the QR code

> To enroll a device using the QR code:

Author: Verify flow once we have an end-to-end demo.

1. [Download](/cloudflare-one/connections/connect-devices/warp/download-warp/) and install the Cloudflare One Agent app.

> 2. Scan the QR code provided by your administrator. Alternatively, go to the login link that is embedded in the QR code.

Reviewer: Should we explicitly document that?

Open the camera app and scan the QR code provided ....

		The Cloudflare One Agent app will open and start the onboarding flow.
3. Review the privacy policy and select **Accept**.
4. On the **Enter team name** screen, confirm that the pre-populated <GlossaryTooltip term="team name">team name</GlossaryTooltip> matches your organization.

		:::note[`Already Authenticated` error]
		If Cloudflare One Agent is already logged in using another team name, you must first log out of that account. Go to **Settings** > **Account** to log out, and then retry the QR code or login link.

> 		:::

Reviewer: Should we also document the case that:

```
If Cloudflare One Agent is already logged in using the same team name retrieved from the QR code, it would launch the Cloudflare One Agent app but not action will be taken.
```

Reviewer 2: adding to Jay's comment.
If Cloudflare One Agent is already logged in using the same team name retrieved from the URL, it would launch the Cloudflare One Agent app but no action will be taken. The user will remain logged in through the current team name.

## iOS, Android, and ChromeOS

### Enroll manually

<Render file="warp/enroll-ios-android" />

The device is now protected by your organization's Zero Trust policies.

### Enroll using a URL <Badge text="Beta" variant="caution"/>

<Details header="Feature availability">

| System   | Availability | Minimum WARP version |
| -------- | ------------ | -------------------- |
| Windows  | ❌           |                      |
| macOS    | ❌           |                      |
| Linux    | ❌           |                      |
| iOS      | ✅           | 1.10                  |
| Android  | ✅           | 2.4                   |
| ChromeOS | ✅           | 2.4                   |

</Details>

> Administrators can provide users with a custom login URL that automatically fills in your organization's <GlossaryTooltip term="team name">team name</GlossaryTooltip> during device enrollment. Using a URL reduces the potential for error that comes with manual entry of the team name. The WARP client supports URLs accessed through a direct link or a QR code.

Author: ```suggestion
Administrators can provide users with a custom login URL that automatically fills in your organization's <GlossaryTooltip term="team name">team name</GlossaryTooltip> during device enrollment. Using a URL reduces the potential for error that comes with manual entry of the team name. The WARP client supports URLs accessed through a direct link or with a URL handler such as a QR code.
```

> 

Author: Direct links are only supported using Safari and Firefox.

#### Generate a login URL

To generate a URL for device enrollment:

> 1. Copy the following link, replacing `your_team_name` with your Zero Trust <GlossaryTooltip term="team name">team name</GlossaryTooltip>:

Author: ```suggestion
1. Copy the following link, replacing `<your-team-name>` with your Zero Trust <GlossaryTooltip term="team name">team name</GlossaryTooltip>:
```

		```txt

> 		cf1app://oneapp.cloudflare.com/team?name=your_team_name

Reviewer 3: Any reason for it being `your_team_name` instead of `<YOUR_TEAM_NAME>`?

> 		cf1app://oneapp.cloudflare.com/team?name=your_team_name

Author: ```suggestion
		cf1app://oneapp.cloudflare.com/team?name={your_team_name}
```

		```

> 2. (Optional) Use any QR code generator to embed the link in a QR code.

Author: ```suggestion
2. (Optional) Use any URL handler, such as a QR code generator, to embed the link.
```

> 3. Distribute the link or QR code to users.

Reviewer: @kokolocomotion1 @ranbel 

2nd step is not optional for the QR code feature. Admin needs to embed the link the QR code. 
3rd step: Admin should distribute the QR code to users (not the URL)

Distributing the link could result in unexpected behavior (like app won't open). Some browser does not redirect to app. But if user clicks the link from the Camera app, the OS/System redirect to the specific app which handles the custom URL.

Reviewer: With further testing and discussion with @kokolocomotion1 -

**iOS**

Entering the `cf1app://oneapp.cloudflare.com/team?name=your_team_name` URL in the address bar of the Safari browser prompts the user to open the app.

**Android**
Entering the `cf1app://oneapp.cloudflare.com/team?name=your_team_name` URL in the address bar of the Chrome browser or Opera Mini does NOT prompt the user to open the app. However, if I use Firefox browser and enter the URL in the address bar it does prompt the user to open the app.

Reviewer: Trying to find if there is any settings on Chrome browser which could enable redirection similar to Firefox.

#### Use the login URL

To enroll a device using a login URL:

1. [Download](/cloudflare-one/connections/connect-devices/warp/download-warp/) and install the Cloudflare One Agent app.

> 2. Go to the [login URL](#generate-a-login-url) provided by your account administrator. Direct links and QR code embedded links are supported. To use a QR code, open the QR scanner app on your device and scan the QR code.

Author: ```suggestion
2. Go to the [login URL](#generate-a-login-url) provided by your account administrator. To use a QR code, open the QR scanner app on your device and scan the QR code.
```

		The Cloudflare One Agent app will open and start the onboarding flow.

		:::note

> 		If the device is already enrolled in this account, Cloudflare One agent will bypass onboarding and show the **Connected** switch.

Author: ```suggestion
		If the device is already enrolled in the account associated with this URL, Cloudflare One agent will bypass onboarding and show the **Connected** switch.
```
